crates/key-value-azure/src/store.rs

```diff
@@ -327,15 +383,18 @@ impl Cas for CompareAndSwap {
                 // attempt to replace the document if the etag matches
                 doc_client
                     .replace_document(pair)
+                    .allow_tentative_writes(TentativeWritesAllowance::Allow)
```

Comment:
I'm a little concerned that by allowing tentative writes, this becomes non-atomic via last in wins default conflict resolution. It seems like this operation should be targeted at the hub rather than one of the spoke db instances. Thoughts?

Reply:
This is a good point. One option would be to remove `allow_tentative_writes` from all atomic operations. Right now, through runtime config, there is no way to configure a target endpoint/region so the hub endpoint is always generated anyways. This brings up the point that we should probably hold off on `allow_tentative_writes` until we update runtime config to support configuring non-primary regions.